libs/utils/graph/serde.rs

```diff
@@ -48,10 +48,9 @@ impl FerriedData {
     pub fn assign_properties(&mut self) {
         let mut to_delete = Vec::new();
         for property in &mut self.unassigned_properties {
-            let node = graph_utils::get_node_with_id(&self.trace_graph, property.entity.clone());
-            if node.is_some() {
+            if let Some(node) = graph_utils::get_node_with_id(&self.trace_graph, property.entity.clone()) {
                 self.trace_graph
-                    .node_weight_mut(node.unwrap())
+                    .node_weight_mut(node)
                     .unwrap()
```

Comment:
`unwrap` in general is an anti-pattern imho because the code crashes without a diagnostic. The more we can get rid of it the better. So for example, we should strive avoid data constructs, where we are forced to use an option. Ideally, we would replace it with [?](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html) for better error handling. Just a general comment.